src/device/msccl_kernel_impl.h

```diff
@@ -49,8 +49,13 @@ extern __shared__ struct mscclShmemData mscclShmem;
 
 inline __device__ static void barrier(int nthreads) {
 #if defined(__HIP_PLATFORM_AMD__) || defined(__HCC__) || defined(__HIPCC__)
-  assert(nthreads == NCCL_MAX_NTHREADS);
-  __asm__ __volatile__("s_waitcnt vmcnt(0) lgkmcnt(0)\ns_barrier");
+  #ifdef __GFX12__
+    assert(nthreads == NCCL_MAX_NTHREADS);
```

Comment:
this line is common for both cases, right?

Reply:
Yes, the spaces are making it look different